homeassistant/components/bring/notify.py

```diff
@@ -0,0 +1,116 @@
+    )
+
+    # registers service bring.send_message with custom service schema.
+    # message is defined as radio input with predefined values, as bring only
+    # accepts 4 predefined values GOING_SHOPPING, CHANGED_LIST, SHOPPING_DONE, URGENT_MESSAGE.
+    # These are like translation_keys and get localized in the app to notification in the users language.
+    # URGENT_MESSAGE also contains a placeholder for item but the notify entity component does not allow
+    # extra data keys yet, so this service is required to be able to invoke URGENT_MESSAGE with the item field.
+
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        "send_message",
+        make_entity_service_schema(
+            {
+                vol.Required(ATTR_NOTIFICATION_TYPE): vol.All(
+                    vol.Upper, cv.enum(BringNotificationType)
+                ),
+                vol.Optional(ATTR_ITEM_NAME): cv.string,
+            }
+        ),
+        "async_send_bring_message",
+    )
+
+
+class BringNotify(NotifyEntity):
+    """Representation of a Bring notify entity."""
+
+    _attr_has_entity_name = False
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: BringDataUpdateCoordinator,
+        bring_list: BringData,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the Notify entity."""
+        if TYPE_CHECKING:
+            assert entry.unique_id
+
+        self.coordinator = coordinator
+        self._list_uuid = bring_list["listUuid"]
+        self._attr_name = bring_list["name"]
+
+        self._attr_unique_id = f"{entry.unique_id}_{self._list_uuid}"
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry.unique_id)},
+            manufacturer=MANUFACTURER,
+            model=SERVICE_NAME,
+        )
+
+    async def async_send_message(self, message: str) -> None:
+        """Send a push notification to members of a To-Do list."""
+        try:
+            await self.async_send_bring_message(message=BringNotificationType[message])
+        except KeyError as e:
+            raise HomeAssistantError(
+                f"Message must contain one of {", ".join(x.value for x in BringNotificationType)}"
+            ) from e
+
+    async def async_send_bring_message(
+        self,
+        message: BringNotificationType,
+        item: str | None = None,
+    ) -> None:
+        """Send a push notification to members of a To-Do list."""
+
+        try:
+            await self.coordinator.bring.notify(self._list_uuid, message, item)
+        except BringRequestException as e:
+            raise HomeAssistantError(
+                "Unable to send push notification for bring"
+            ) from e
+        except ValueError as e:
+            raise HomeAssistantError(
+                "Item name is required for Breaking news notification"
+            ) from e
```

Comment:
Raise `ServiceValudationError` and move the error message to the translations (`string.json`)